src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,74 @@
 package core.basesyntax;
 
+import java.io.*;
+
 public class WorkWithFile {
+    private static final int TRANSACTION_TYPE_INDEX = 1;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeTextToFile(
+                createReport(getTextFromFile(fromFileName)),
+                toFileName);
+
+    }
```

Comment:
The method 'getStatistic' is doing more than one operation. It's getting the text from a file, creating a report, and writing to another file. Consider splitting this into separate methods to adhere to the single responsibility principle.